src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Setting/WritableSettingsStoreWrapper.cs

```diff
@@ -20,32 +20,62 @@ public WritableSettingsStoreWrapper(IVsWritableSettingsStore store)
 
         public void DeleteCollection(string collection)
         {
-            _store.DeleteCollection(collection);
+            NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
+            {
+                await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+                _store.DeleteCollection(collection);
```

Comment:
>_store.DeleteCollection [](start = 16, length = 23)

Here and everywhere: use `ErrorHandler.ThrowOnFailure( )` around all these HRESULT returning calls or you'll ignore any failures that in CLR land would have thrown exceptions.

Reply:
makes sense, but I don't wana do that as part of this PR. I'll create another tracking workitem to do this across the code base.